installers/BuildFromSource.hs

```diff
@@ -127,22 +127,32 @@ makeRepos artifactDirectory repos =
  do createDirectoryIfMissing True artifactDirectory
     setCurrentDirectory artifactDirectory
     root <- getCurrentDirectory
-    cabal [ "sandbox", "init" ]
     mapM_ (uncurry (makeRepo root)) repos
 
+    -- create a sandbox for installation
+    cabal [ "sandbox", "init" ]
+
+    -- add each of the sub-directories as a sandbox source
+    cabal ([ "sandbox", "add-source" ] ++ map fst repos)
+    
+    -- install all of the packages together in order to resolve transitive dependencies robustly
+    -- (install the dependencies a bit more quietly than the elm packages)
+    cabal ([ "install", "-j", "--only-dependencies", "--ghc-options=\"-w\"" ] ++ map fst repos)
+    cabal ([ "install", "-j", "--bindir=../bin", "--ghc-options=\"-XFlexibleContexts\"" ] ++ filter (/= "elm-reactor") (map fst repos))
```

Comment:
Oh, it's definitely GHC 7.10, it says so in the [commit where it's introduced](https://github.com/rehno-lindeque/elm-platform/commit/ef7511453a8e45c48b0c3b80b31e9cfde1790d1f)